internal/db/embedding.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2025 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package db
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/philippgille/chromem-go"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/errors"
+	"github.com/sourcenetwork/defradb/internal/keys"
+)
+
+const (
+	ollama string = "ollama"
+	openai string = "openai"
+)
+
+var supportedEmbeddingProviders = map[string]struct{}{
```

Comment:
nitpick: Using 'vector embedding' terminology over just 'embedding' here and other places in this file.

Reply:
Noted. I'll fix it in the _similarity PR coming soon.